Source/WebCore/css/StyleProperties.cpp

```diff
@@ -609,24 +690,21 @@ String StyleProperties::fontValue() const
         }
     }
 
-    auto fontSizeString = fontSizeProperty.value()->cssText();
-    auto commonValue = fontSizeString;
+    // This code no longer uses commonValue, but we define it so we can use appendFontLonghandValueIfExplicit.
```

Comment:
I find this a bit unfortunate, even if we save a bit of string builder work, since we now have the set of property names repeated 3 times.

Reply:
I think that we can and will work on making this better over time. The special considerations for various longhands makes it hard to drive this from a single list of properties, but I won’t give up on that. For example, when actually serializing `font-stretch` needs lots of special handling, but the rules about CSS-wide keywords are the same as for every other property.

It’s a bad pattern to first call `cssText` and then after the fact check if the string happens to be a CSS-wide keyword. We should get rid of that completely. But it’s also bad to look up the same property more than once.

We need to do some of these same kinds of checks for `fontVariantValue`, so then the `appendFontLonghandValueIfExplicit` function can be changed too since that is the only other function that uses it.

It would be nice if this function could share lists of properties with the `consumeFont` function. I tried to take a step in that direction when rewriting `StyleProperties::isSingleFontShorthand` and it made the function much shorter too!

Reply:
The whole `commonValue` idiom doesn’t really work, and is based on false ideas about the task. Mon-CSS-wide-keyword values shouldn’t be checked for "common" values, it’s just a waste. And keywords like `inherit` can’t appear as values for single items within the shorthand, so they need to be handled differently whether they are "common" or not. This is handled wrong in the other shorthands, and is part of what we need to fix.